### PR TITLE
cpu/native: remove unused can_linux module

### PR DESCRIPTION
**Pull request by leandrolanzieri**
### Contribution description
b4f29035ce6ee12949df1227ee463340ea8e8571 adapted the can_linux module to the periph_can interface. This is a cleanup of some things that stayed behind. Here the makefile is removed, the references to can_linux in the dependency resolution and configuration Makefile are changed to the standard periph_can, and the startup code is adapted.

### Testing procedure
- `test/candev` when using periph_can should still work on native (I'm not able to test this locally)

### Issues/PRs references
Module modified in #14909

### CODE REVIEW

File: cpu/native/periph/can.c [hunk 1]
```diff
@@ -20,7 +20,7 @@
  */
 
 #if !defined(__linux__)
-#error "MODULE can_linux is only available on Linux"
+#error "MODULE periph_can is only available on Linux"
```
Comment (wosym) [inline cpu/native/periph/can.c:23]: I currently can't check, but OTOMH this is not correct. 
Many controllers (stm32 for example) also offer a can peripheral.

Comment (leandrolanzieri) [reply cpu/native/periph/can.c:23]: This error is in the `native` can implementation...

Comment (wosym) [reply cpu/native/periph/can.c:23]: Yes, but since the native can implementation now uses the same can implementation as can peripherals (periph_can), the statement "MODULE periph_can is only available on Linux" doesn't sound right?

Comment (leandrolanzieri) [reply cpu/native/periph/can.c:23]: This message is there for when trying to use the `periph_can` on a native instance that is not linux (e.g. on OSX), although I'd say that in that case the features should not even be there. Keep in mind that this file is `cpu/native` specific.

> the statement "MODULE periph_can is only available on Linux" doesn't sound right?

How do you propose to reword this?

Comment (wosym) [reply cpu/native/periph/can.c:23]: I understand the purpose of this message (even though at first I also misinterpreted it), but for people unfamiliar to RIOT or unfamiliar to the RIOT can-system, this message is very ambiguous. It can be understood as "You can only use periph_can when running RIOT on Linux (i.e. native)", which is not the case. 
I would reword it to something along the lines of "Usage of periph_can on RIOT native requires libsocketcan. Currently this is only available on Linux". 

(I'm still not 100% convinced about my own proposition. There's still room for confusion. But I would go more or less in this direction to avoid users from misinterpreting what is supported and what is not.)

Comment (leandrolanzieri) [reply cpu/native/periph/can.c]: I've adapted the message

**Review by aabadie — APPROVED** [review]
Looks good and makes sense. ACK, please squash!

**Review by vincent-d — APPROVED** [review]
Looks good. ACK

Thanks @leandrolanzieri

**Comment by leandrolanzieri** [comment]
Thanks everyone for reviewing!
